torch/_inductor/kernel/flex/flex_attention.py

```diff
@@ -212,6 +215,9 @@ def flex_attention(
     score_mod_other_buffers = maybe_realize(score_mod_other_buffers)
```

Comment:
so we have something similar ish w/ realizing input nodes, for my own understanding even though we realize these buffers it is possible that the layouts for these nodes are changed but subsequent nodes?

Reply:
In inductor you have 3 levels:
1) Unrealized: no buffer is allocated usages are inlined into users
2) Realized (FlexibleLayout): a buffer is allocated, but the strides can be changed by optimizations
3) Realized (FrozenLayout): a buffer is allocated, with fixed strides

`.realize()` is about going from 1 to 2.   `.freeze_layout()` goes from 2 to 3.